src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -96,18 +101,76 @@ public function vote(TokenInterface $token, $object, array $attributes)
     }
 
     /**
+     * Determines if the attribute and class are supported by this voter.
+     *
+     * To determine if the passed class is instance of the supported class, the
+     * isClassInstanceOf() method can be used.
+     *
+     * @param string $attribute An attribute
+     * @param string $class     The fully qualified class name of the passed object
+     *
+     * @return bool True if the attribute and class is supported, false otherwise
+     */
+    protected function supports($attribute, $class)
```

Comment:
yes it will. And this is why there is a deprecation warning in the provided implementation.

@WouterJ please add a comment in the phpdoc saying it will become abstract in 3.0